samcli/local/lambda_service/service.py

```diff
@@ -0,0 +1,128 @@
+"""Local Lambda Service"""
+
+import logging
+import re
+import io
+import os
+
+from flask import Flask, request
+
+
+from samcli.local.services.base_service import BaseService
+from samcli.local.lambdafn.exceptions import FunctionNotFound
+
+LOG = logging.getLogger(__name__)
+
+
+class LocalLambdaService(BaseService):
+
+    _DEFAULT_PORT = 3001
+    _DEFAULT_HOST = '127.0.0.1'
+
+    def __init__(self, function_name_list, lambda_runner, port=None, host=None, stderr=None):
+        """
+        Creates a Local Lambda Service
+
+        Parameters
+        ----------
+        function_name_list list of str
+            List of the Function Logical Ids
+        lambda_runner samcli.commands.local.lib.local_lambda.LocalLambdaRunner
+            The Lambda runner class capable of invoking the function
+        port int
+            Optional. port for the service to start listening on. Defaults to 3001
+        host str
+            Optional. host to start the service on Defaults to '127.0.0.1'
+        stderr io.BaseIO
+            Optional stream where the stderr from Docker container should be written to
+        """
+        self.function_name_list = function_name_list
+        super(LocalLambdaService, self).__init__(lambda_runner, port=port, host=host, stderr=stderr)
+
+    def create(self):
+        """
+        Creates a Flask Application that can be started.
+        """
+        self._app = Flask(__name__)
+
+        for function_name in self.function_name_list:
+            path = '/2015-03-31/functions/{}/invocations'.format(function_name)
+            self._app.add_url_rule(path,
+                                   endpoint=path,
+                                   view_func=self._invoke_request_handler,
+                                   methods=['POST'],
+                                   provide_automatic_options=False)
+
+        self._construct_error_handling()
+
+    def run(self):
+        """
+        This starts up the (threaded) Local Server.
+        Note: This is a **blocking call**
+
+        Raises
+        ------
+        RuntimeError
+            if the service was not created
+        """
+        if not self._app:
+            raise RuntimeError("The application must be created before running")
+
+        # Flask can operate as a single threaded server (which is default) and a multi-threaded server which is
```

Comment:
so this logic is spread in two classes now? This one and the APIGW Service?

Reply:
Ya. I didn't get far enough to get this abstracted yet but it totally should be. Will work on getting that and cleaning up the current 'Base' class.